writegood-mode.el

```diff
@@ -272,17 +278,104 @@ From Wikipedia URL `https://en.wikipedia.org/wiki/Flesch–Kincaid_readability_t
+;;; BEGIN Score on the fly
+(defun writegood-calculate-reading-ease (&optional start end)
+  "Calculate score of Flesch-Kincaid reading ease test in the region bounded by START and END.
 
 Scores roughly between 0 and 100."
-   (interactive)
    (let* ((params (writegood-fk-parameters start end))
           (sentences (nth 0 params))
           (words     (nth 1 params))
           (syllables (nth 2 params))
           (score  (- 206.835 (* 1.015 (/ words sentences)) (* 84.6 (/ syllables words)))))
+     score))
+
+(defun writegoodmode-reading-ease-thing-at-point (thing)
+  "Calculate score for thing at point."
+  (let* ((bounds (bounds-of-thing-at-point thing))
+         (b (car bounds))
+         (e (cdr bounds)))
+    (if (and
+         (not (null b))
+         (not (null e))
+         ;; this is a guess: when the interval between boundaries is
+         ;; huge, the paragraph is too big to be validated.
+         (< (- e b) 100000))
+        (let ((score (writegood-calculate-reading-ease b e)))
+          (message "%s reading ease score: %.2f %s" (symbol-name thing) score
+            (writegood-reading-ease-score->comment score))))))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-sentence ()
+  "Calculate score for the sentence at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'sentence))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-paragraph ()
+  "Calculate score for the paragraph at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'paragraph))
+
+;;;###autoload
+(defun writegoodmode-reading-ease-page ()
+  "Calculate score for the page at point."
+  (interactive)
+  (writegoodmode-reading-ease-thing-at-point 'page))
+
+(defun writegood-after-sentence ()
+  "Calculate reading ease after a sentence is completed. Here we
+consider the sentence completion the addition of a dot."
+  (if (and (string-match-p (regexp-quote "self-insert-command") (symbol-name last-command)) (eq (char-before) 46)) (writegoodmode-reading-ease-sentence)))
+
+(defun writegood-after-paragraph ()
+  "Calculate reading ease after a paragraph is completed. Here we
+consider the paragraph completion to be the call of a
+,*-fill-paragraph command."
+  (if (string-match-p (regexp-quote "fill-paragraph") (symbol-name last-command)) (writegoodmode-reading-ease-paragraph)))
+
+(defun apply-only-in-text-major-modes (fn)
+  (if (member (symbol-name major-mode) writegood-on-the-fly-modes)
+      (funcall fn)))
+
+(defun writegood-after-sentence-hook () (apply-only-in-text-major-modes 'writegood-after-sentence))
+(defun writegood-after-paragraph-hook () (apply-only-in-text-major-modes 'writegood-after-paragraph))
+(defun writegood-after-page-hook () (apply-only-in-text-major-modes 'writegoodmode-reading-ease-page))
+
+(defun writegood-on-the-fly-turn-on ()
+  "Add hooks to enable on-the-fly scoring."
+  (add-hook 'post-command-hook 'writegood-after-sentence-hook)
```

Comment:
I think we can probably use something like this instead:

```lisp
(add-hook 'post-self-insert-hook 'syntax-check-continuous nil t)
```

Really, the `post-self-insert-hook` is the part we want. That will take away the need to string-match the command name. We'll still need to check for a sentence end.

Reply:
This sounds interesting! I will play around with it and propose an alternative.